Dependency on: gdb-6.5-bz185337-resolve-tls-without-debuginfo-v2.patch


2006-09-01  Jan Kratochvil  <jan.kratochvil@redhat.com>

	* target.c (target_translate_tls_address): Fix for separate debuginfo.


diff -rup gdb-6.5.orig/gdb/target.c gdb-6.5/gdb/target.c
--- gdb-6.5.orig/gdb/target.c	2006-09-20 17:13:35.000000000 -0400
+++ gdb-6.5/gdb/target.c	2006-09-20 17:15:53.000000000 -0400
@@ -769,6 +769,12 @@ target_translate_tls_address (struct obj
       ptid_t ptid = inferior_ptid;
       volatile struct gdb_exception ex;
 
+      /* Resolve: Cannot find shared library
+	 `/usr/lib/debug/lib/lib....so.debug' in dynamic linker's load
+	 module list  */
+      if (objfile->separate_debug_objfile_backlink != NULL)
+	objfile = objfile->separate_debug_objfile_backlink;
+
       TRY_CATCH (ex, RETURN_MASK_ALL)
 	{
 	  CORE_ADDR lm_addr;
